homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -35,6 +48,39 @@
 
 SUCCESS = ["ok"]
 
+GATEWAY_SENSOR_TYPES = {
+    "temperature": [TEMP_CELSIUS, None, DEVICE_CLASS_TEMPERATURE],
```

Comment:
did it

Reply:
Looks much better! :+1: But why are the icons always `None`?

Reply:
I actually based this code on the xiaomi_aqara integration and it was done in this way.
Because the DEVICE_CLASS is already set the icon will be set automattically and there is no need to specify the icon explicitly. That is why None is simply passed.
However the temperature senser will have the temperature icon as set by the DEVICE_CLASS_TEMPERATURE. I don't want to explicitly state the icon because if HomeAssistant decides the icon needs to change for the device class this integration schould change with it.

So bassically I could just remove the icon all together, but I kept it because there are some sensors that do not have a HomeAssistant Device Class. For example the xiaomi aqara vibration sensor does not fit in any of the HomeAssistant Device Classes, so for the vibration sensor we could define or own icon in this way.

Reply:
Thanks for the comprehensive elaboration, I didn't know device class handles that all automatically, but the it makes sense to me to have it there for the future extensions.